src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -203,14 +201,24 @@ export class LocalKernelFinder implements ILocalKernelFinder {
                     if (
                         kernelspec.language === PYTHON_LANGUAGE &&
                         hideDefaultKernelSpecs &&
-                        kernelspec.name.toLowerCase().match(isDefaultPythonKernelSpecName)
+                        (kernelspec.name.toLowerCase().match(isDefaultPythonKernelSpecName) ||
+                            kernelspec.argv[0].toLowerCase() === 'python' ||
+                            kernelspec.argv[0].toLowerCase().startsWith('/usr/bin/python'))
                     ) {
+                        traceInfo(`Hiding default kernel spec ${kernelspec.display_name}, ${kernelspec.argv[0]}`);
                         return false;
                     }
                     // Disable xeus python for now.
                     if (kernelspec.argv[0].toLowerCase().endsWith('xpython')) {
+                        traceInfo(`Hiding xeus kernelspec`);
+                        return false;
+                    }
+                    const extensionId = kernelspec.metadata?.vscode?.extension_id;
+                    if (extensionId && this.extensions.getExtension(extensionId)) {
+                        traceInfo(`Hiding kernelspec ${kernelspec.display_name}, better support by ${extensionId}`);
```

Comment:
Moved this from the controller manager into this place (we hide default kernels, xeus kernel in `localKernelFinder.ts`, hecne moved the ability to hide kernels that have corresponding VSCode Extensions.

Reply:
Doesn't this mean .NET won't show up for stable then? Actually I guess it doesn't matter because if .NET is installed we use native anyway.